lib/Wikia/src/Tracer/WikiaTracer.php

```diff
@@ -52,7 +53,8 @@ private function __construct() {
 		$this->clientIp = $this->getTraceEntry( self::CLIENT_IP_HEADER_NAME );
 		$this->clientBeaconId = $this->getTraceEntry( self::CLIENT_BEACON_ID_HEADER_NAME );
 		$this->clientDeviceId = $this->getTraceEntry( self::CLIENT_DEVICE_ID_HEADER_NAME );
-		$this->userId = $this->getTraceEntry( self::CLIENT_USER_ID );
+		$this->userId = $this->getTraceEntry( self::CLIENT_USER_ID )
+			?: $this->getTraceEntry( self::LEGACY_CLIENT_USER_ID );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)